### ISSUE TITLE
Better provide a docker image for building

### ISSUE THREAD

**Issue by Shylock-Hg** [vesoft-inc/nebula#877]
**Is your feature request related to a problem? Please describe.**
The building method in document is not usable sometime.The docker image for building will
be better to build.

**Describe the solution you'd like**
Provide the docker image with basic and thirdparty library, build tools  installed.

**Describe alternatives you've considered**
...

**Additional context**
...

**Comment by laura-ding** [comment]
[image](https://user-images.githubusercontent.com/48548375/64073032-33bd9280-cccb-11e9-8bbd-ac89375f97cf.png)

3rd-party does not support `split compilation`,  cmake directly under `nebula-3rdparty/`

Thanks very much for your advice. We have work in process.

**Comment by Shylock-Hg** [comment]
> ![image](https://user-images.githubusercontent.com/48548375/64073172-bdba2b00-cccc-11e9-8281-4c77fbc0b917.png)
> 
> 3rd-party does not support `split compilation`, cmake directly under `nebula-3rdparty/`
> 
> Thanks very much for your advice. We have work in process.

Yes, I had tried this and disable parallel compilation,but not work.

The log:
Cloning into 'nebula-3rdparty'...
-- The C compiler identification is GNU 8.1.0
-- The CXX compiler identification is GNU 8.1.0
-- Check for working C compiler: /home/engshare/gcc/bin/gcc
-- Check for working C compiler: /home/engshare/gcc/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /home/engshare/gcc/bin/g++
-- Check for working CXX compiler: /home/engshare/gcc/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- SKIP_JAVA_JAR = ON
-- Set macro _FORTIFY_SOURCE=2
-- CMAKE_BUILD_TYPE = Release (Options are: Debug, Release, RelWithDebInfo, MinSizeRel)
-- THIRDPARTY_PATH = /opt/nebula/third-party
-- Specified NEBULA_KRB5_ROOT: /home/engshare/krb5
-- Specified NEBULA_LIBUNWIND_ROOT: /home/engshare/libunwind
-- Specified NEBULA_OPENSSL_ROOT: /home/engshare/openssl
-- Specified NEBULA_FLEX_ROOT: /home/engshare/flex
-- Specified NEBULA_GPERF_BIN_DIR: /home/engshare/gperf/bin
-- CMAKE_INCLUDE_PATH: /home/engshare/flex/include:/home/engshare/boost/include:/home/engshare/openssl/include:/home/engshare/libunwind/include:/home/engshare/krb5/include
-- CMAKE_LIBRARY_PATH: /home/engshare/flex/lib:/home/engshare/boost/lib:/home/engshare/openssl/lib:/home/engshare/libunwind/lib:/home/engshare/krb5/lib
-- CMAKE_PROGRAM_PATH: /home/engshare/gperf/bin:/home/engshare/flex/bin:/home/engshare/krb5/bin
-- Boost version: 1.67.0
-- Found OpenSSL: /home/engshare/openssl/lib/libcrypto.a (found version "1.1.0h") 
-- found krb5-config here /home/engshare/krb5/bin/krb5-config
-- Found kerberos 5 headers: /home/engshare/krb5/include
-- Found kerberos 5 libs:    /home/engshare/krb5/lib/libgssapi_krb5.a;/home/engshare/krb5/lib/libkrb5.a;/home/engshare/krb5/lib/libk5crypto.a;/home/engshare/krb5/lib/libcom_err.a;/home/engshare/krb5/lib/libkrb5support.a
-- Found GPERF: /home/engshare/gperf/bin/gperf (found suitable version "3.1", minimum required is "2.8") 
-- Found Libunwind: /home/engshare/libunwind/lib/libunwind.a  
-- Found BISON: /usr/bin/bison (found version "3.0.4") 
-- Found FLEX: /home/engshare/flex/bin/flex (found version "2.6.4") 
-- Found ncurses: /usr/include  
-- Looking for lzma_auto_decoder in /usr/lib64/liblzma.so
-- Looking for lzma_auto_decoder in /usr/lib64/liblzma.so - found
-- Looking for lzma_easy_encoder in /usr/lib64/liblzma.so
-- Looking for lzma_easy_encoder in /usr/lib64/liblzma.so - found
-- Looking for lzma_lzma_preset in /usr/lib64/liblzma.so
-- Looking for lzma_lzma_preset in /usr/lib64/liblzma.so - found
-- Found LibLZMA: /usr/include (found version "5.2.2") 
-- Glibc version is 2.17
-- Configuring done
-- Generating done
-- Build files have been written to: /root/nebula-3rdparty
Scanning dependencies of target tgt_zstd
make[2]: *** [third-party/zstd/CMakeFiles/tgt_zstd] Error 1
make[1]: *** [third-party/zstd/CMakeFiles/tgt_zstd.dir/all] Error 2
make: *** [all] Error 2
The command '/bin/sh -c cd && git clone --depth=1 https://github.com/vesoft-inc/nebula-3rdparty.git &&     cd nebula-3rdparty &&     /home/engshare/cmake/bin/cmake -DCMAKE_C_COMPILER=/home/engshare/gcc/bin/gcc -DCMAKE_CXX_COMPILER=/home/engshare/gcc/bin/g++ -DNEBULA_GPERF_BIN_DIR=/home/engshare/gperf/bin -DNEBULA_FLEX_ROOT=/home/engshare/flex -DNEBULA_BOOST_ROOT=/home/engshare/boost -DNEBULA_OPENSSL_ROOT=/home/engshare/openssl -DNEBULA_KRB5_ROOT=/home/engshare/krb5 -DNEBULA_LIBUNWIND_ROOT=/home/engshare/libunwind ./ &&     make && make install && cd && rm -rf nebula-3rdparty' returned a non-zero code: 2

**Comment by whitewum** [comment]
Right. We‘re working on the docker now. Thank you for your suggestion.

**Comment by laura-ding** [comment]
![image](https://user-images.githubusercontent.com/48548375/64073906-e34c3200-ccd6-11e9-8557-b9b4132355a7.png)
Please do this

**Comment by Shylock-Hg** [comment]
> ![image](https://user-images.githubusercontent.com/48548375/64073906-e34c3200-ccd6-11e9-8557-b9b4132355a7.png)
> Please do this

It's seems ok. But the quota of lfs limited...

batch response: This repository is over its data quota. Purchase more data packs to restore access.

**Comment by laura-ding** [comment]
Thanks for your feedback. We will solve this problem as soon as possible.

**Comment by yixinglu** [comment]
I am focusing on this issue. The solution is work in progress. We will publish nebula docker images address for developers as soon as possible.

**Comment by laura-ding** [comment]
Please delete the download and clone `nebula -3rdparty` again.
Thank you very much!

**Comment by yixinglu** [comment]
Nebula development docker image is ready now. Welcome to try it out from [docker hub](https://hub.docker.com/r/vesoft/nebula-dev).

We could close this issue now. @laura-ding
